python/ray/actor.py

```diff
@@ -613,6 +613,41 @@ def remote(self, *args, **kwargs):
                     **cls_options,
                 )
 
+            def get_or_create(self, *args, **kwargs):
+                """Get or create a named actor.
+
+                The name of the actor must be set via `Actor.options(name="name")`
+                prior to calling this method.
+
+                If the actor already exists, a handle to the actor will be returned
+                and the arguments will be ignored. Otherwise, a new actor will be
+                created with the specified arguments.
+
+                Args:
+                    args: These arguments are forwarded directly to the actor
+                        constructor.
+                    kwargs: These arguments are forwarded directly to the actor
+                        constructor.
+
+                Returns:
+                    A handle to the new or existing named actor.
+                """
+                if not cls_options.get("name"):
+                    raise ValueError(
+                        "The actor name must be specified before use `get_or_create`."
+                    )
+                name = cls_options["name"]
+                try:
+                    return ray.get_actor(name, namespace=cls_options.get("namespace"))
+                except ValueError:
+                    # Attempt to create it (may race with other attempts).
+                    try:
+                        return self.remote(*args, **kwargs)
+                    except ValueError:
+                        # We lost the creation race, ignore.
+                        pass
```

Comment:
is it worth logging a warning in this case?

Reply:
It's expected, so we shouldn't log anything (people have complained about this before when we force-logged warnings in core worker).

Reply:
fair point; ideally we'd use [warnings](https://docs.python.org/3/library/warnings.html) everywhere so people could suppress things as desired